### DISCUSSION TITLE
Where can I translate the table

### DISCUSSION THREAD

**Discussion by Todomed** [rappasoft/laravel-livewire-tables#575] [Q&A]
Can you guide me where I can translate the table and an example to activate bulk actions and columns

**Comment by juanzu** [comment]
> Can you guide me where I can translate the table and an example to activate bulk actions and columns

buddy, did you figure it out?

**Comment by devmikromike** [comment]
Any luck with this?, anyone ?

**Comment by lrljoe** [comment]
Can you expand on what you're trying to achieve?

There are translations for quite a few languages built in already, is there a particular piece of text that isn't translating?

For Bulk Actions, there's the "Advanced Example" available which outlines how to achieve it, if there's something unclear, please let me know what it is that's unclear and I'll explain and update the docs.

**Comment by devmikromike** [comment]
Not in Finnish, I have publiched vendor, translations goes to lang/vendor/livewire-tables.

If I  add fi.json file , it is not translate UI elements, like search or Colums.
ex:    "Search": "Haku",

**Reply by lrljoe** [reply]
I'm assuming that the rest of your site is translating correctly into Finnish, and you've set fi as your default in your app?

If you don't mind sharing your Finnish translation file, then I can take a quick look in my demo environment, and I can also get it added into a future release to make life easier for you too.

**Reply by devmikromike** [reply]
I can share translation, do I open pr with translation file ?

**Reply by lrljoe** [reply]
You certainly can do, and I know it'll be appreciated.  Let's try and figure out why it's not working first though!

Have you set the your default locale?

If you add the following to the top of your Table Component file:
```php
use Illuminate\Support\Facades\App;
```

And add the following to the configure() method:
```php
App::setLocale('fi');
```

Does it start using your Finnish translation?

**Reply by devmikromike** [reply]
by adding this line use Illuminate\Support\Facades\App; start working, thanks.

Using Laravel 9.19.

**Reply by lrljoe** [reply]
Ok sounds like a problem somewhere other than this component!

However, still happy to help.

Do you only have Finnish users, or are you needing localisation per-user?

In your config/app.php is Finnish set as the default locale?  It defaults to English otherwise.  Be aware this will affect your whole site.

Are you using any middleware to set the locale?

**Comment by devmikromike** [comment]
Well in this case, there is only one end user.

Yes, it is not component problem, maybe bug in Laravel version.
Anyway, I have translation file for you, if  you want it.

But there  is Finnish language issue.

English word “results” is translated differently in different situations like:

Showing 1 to 25 of 90 results  = “näyttää 1:stä 25:een, 90:stä vastauksesta”

So we add word ”to” end of words.

And search results:

Showing 0 results = “ näyttää nolla vastausta”

If you want help with, I am happy to help you.

Please let me know.

thanks Mika.

**Reply by lrljoe** [reply]
Absolutely please do issue a PR for the translation, Finnish is not my strong point!

I suspect the localisation/translation issue is within your 
config/app.php
Ensure you have both of the following in the array (so that it will fallback to Finnish)

```php
'locale' => 'fi',
'fallback_locale' => 'fi',
```

It may be worth looking at the CodeZero middleware if you do decide to have multi-lingual, it has lots of customisations for determining locale.

**Comment by lrljoe** [comment]
Finnish translation is in the next release, thank you very much for the contribution!
